tests/v2/validation/upgrade/kubernetes.go

```diff
@@ -1,151 +1,188 @@
-	isCheckingCurrentCluster = true
-	// logMessageKubernetesVersion is the string message to log with the kubernetes versions validations' assertions
-	logMessageKubernetesVersion = "Validating the current version is the upgraded one"
-	// logMessageNodepoolVersion is the string message to log the with kubernetes nodepools versions validations' assertions
-	logMessageNodepoolVersions = "Validating the current nodepools version is the upgraded one"
+	local     = "local"
+	namespace = "fleet-default"
+	provider  = "provider.cattle.io"
+	rke       = "rke"
+	rke2      = "rke2"
 )
 
-func waitUntilLocalStable(client *rancher.Client, clusterName string) error {
-	return kwait.Poll(10*time.Second, 75*time.Minute, func() (bool, error) {
-		isConnected, err := client.IsConnected()
-		if err != nil {
-			logrus.Errorf("[connection stable]: %v", err)
-			return false, nil
-		}
-		logrus.Infof("[connection stable]: %v", isConnected)
+// upgradeLocalCluster is a function to upgrade a local cluster.
+func upgradeLocalCluster(u *suite.Suite, testName string, client *rancher.Client, clusterName string, testConfig *clusters.ClusterConfig, cluster upgradeinput.Cluster) {
+	clusterObject, err := clusters.GetClusterIDByName(client, clusterName)
+	require.NoError(u.T(), err)
+
+	clusterResp, err := client.Management.Cluster.ByID(clusterObject)
+	require.NoError(u.T(), err)
+
+	if clusterResp.Labels[provider] == rke {
+		testConfig.KubernetesVersion = cluster.ProvisioningInput.RKE1KubernetesVersions[0]
+		testName += "Local cluster to " + testConfig.KubernetesVersion
+	} else if clusterResp.Labels[provider] == rke2 {
+		testConfig.KubernetesVersion = cluster.ProvisioningInput.RKE2KubernetesVersions[0]
+		testName += "Local cluster to " + testConfig.KubernetesVersion
+	} else {
+		testConfig.KubernetesVersion = cluster.ProvisioningInput.K3SKubernetesVersions[0]
+		testName += "Local cluster to " + testConfig.KubernetesVersion
+	}
 
-		if isConnected {
-			ready, err := kcluster.IsClusterActive(client, clusterName)
-			if err != nil {
-				logrus.Errorf("[cluster ready]: %v", err)
-				return false, nil
-			}
-			logrus.Infof("[cluster ready]: %v", ready)
+	u.Run(testName, func() {
+		createPreUpgradeWorkloads(u.T(), client, clusterName, cluster.FeaturesToTest)
 
-			return ready, nil
-		}
+		clusterMeta, err := clusters.NewClusterMeta(client, clusterName)
+		require.NoError(u.T(), err)
 
-		return false, nil
+		initCluster, err := bundledclusters.NewWithClusterMeta(clusterMeta)
+		require.NoError(u.T(), err)
+
+		initClusterResp, err := initCluster.Get(client)
+		require.NoError(u.T(), err)
+
+		logrus.Infof("Upgrading local cluster...")
+		_, err = initClusterResp.UpdateKubernetesVersion(client, &testConfig.KubernetesVersion)
+		require.NoError(u.T(), err)
+
+		err = waitForLocalClusterUpgrade(client, clusterName)
+		require.NoError(u.T(), err)
+
+		logrus.Infof("Local cluster has been upgraded!")
```

Comment:
it would be nice instead to have an additional require and log that the k8s version is now the expected (updated) version

Reply:
Resolved